equinox/nn/conv.py

```diff
@@ -0,0 +1,227 @@
+import collections
+import math
+from itertools import repeat
+from typing import Optional, Sequence, Union
+
+import jax.numpy as jnp
+import jax.random as jrandom
+from jax.lax import conv_general_dilated
+
+from ..custom_types import Array
+from ..module import Module, static_field
+
+
+def _ntuple(n):
+    def parse(x):
+        if isinstance(x, collections.abc.Iterable):
+            return tuple(x)
```

Comment:
Would it make sense to check that `x` is of length `n` here? (And raise a `ValueError` if not.)

Reply:
Nice catch. I borrowed this from PyTorch - looks like no one spotted the problem there yet!